### DISCUSSION TITLE
For recorder, can you change the chart type? E.g. to a vertical amplitude bar?

### DISCUSSION THREAD

**Discussion by alankent** [katspaugh/wavesurfer.js#3165] [Q&A] [answered]
In the recorder, can you make it show a different graphical representation? E.g. I was wondering about this sort of vertical representation.

<img width="49" alt="image" src="https://github.com/katspaugh/wavesurfer.js/assets/5702163/79f0f3c9-509b-4cb6-8f06-dea17b3668f2">

Most people don't want to see the actual waveform, but are interested if they are speaking loud enough or close enough (but not too close) to the microphone to get the levels optimal without distortion etc.

And yes, those representations have a bit of "memory" so the peaks decay a bit more slowly so you can see them. No idea how to implement things like that, but it is very easy to ask! ;-)

**Comment by katspaugh** [comment, marked as answer]
That makes sense!
This can be done by providing a custom `renderFunction`, like in this example: https://wavesurfer-js.org/examples/?custom-render.js

Edit: here's a crude volume meter to get you started: https://wavesurfer-js.org/examples/?2528031296bc33df1e07fb2573042d7a

**Comment by alankent** [comment]
I gave that a go and got it rotated to be vertical. My challenge now is I have it in a resizable panel, but the canvas does not seem to resize so the bar height does not change. 

To illustrate: initial size.
<img width="299" alt="image" src="https://github.com/katspaugh/wavesurfer.js/assets/5702163/27da72ed-5b88-4deb-867c-a95e2e19c8ba">

After I stretch the panel taller...

<img width="298" alt="image" src="https://github.com/katspaugh/wavesurfer.js/assets/5702163/09a13ce9-8e92-437d-81ba-c4fe144c652e">

The area for the bar is taller, but it still draws from the initial height. (I am using a react library for resizable panels. I don't know how it exactly works, but I got the video to resize to fit in the available space with the bar next to it)
